test/integrationTests/reAnalyze.integration.test.ts

```diff
@@ -0,0 +1,96 @@
+/*--------------------------------------------------------------------------------------------- 
+*  Copyright (c) Microsoft Corporation. All rights reserved. 
+*  Licensed under the MIT License. See License.txt in the project root for license information. 
+*--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+
+import { should, expect } from 'chai';
+import { activateCSharpExtension } from './integrationHelpers';
+import testAssetWorkspace from './testAssets/testAssetWorkspace';
+import poll from './poll';
+import { EventStream } from '../../src/EventStream';
+import { EventType } from '../../src/omnisharp/EventType';
+import { BaseEvent, OmnisharpProjectDiagnosticStatus } from '../../src/omnisharp/loggingEvents';
+import { DiagnosticStatus } from '../../src/omnisharp/protocol';
+
+const chai = require('chai');
+chai.use(require('chai-arrays'));
+chai.use(require('chai-fs'));
+
+function listenEvents<T extends BaseEvent>(stream: EventStream, type: EventType): T[]
+{
+    let results: T[] = [];
+
+    stream.subscribe((event: BaseEvent) => {
+        if(event.type === type)
+        {
+            results.push(<T>event);
+        }
+    });
+
+    return results;
+}
+
+suite(`ReAnalyze: ${testAssetWorkspace.description}`, function () {
+    let interfaceUri: vscode.Uri;
+    let interfaceImplUri: vscode.Uri;
+    let eventStream: EventStream;
+
+    suiteSetup(async function () {
+        should();
+
+        await testAssetWorkspace.restore();
+        eventStream = (await activateCSharpExtension()).eventStream;
+
+        let projectDirectory = testAssetWorkspace.projects[0].projectDirectoryPath;
+        interfaceUri = vscode.Uri.file(path.join(projectDirectory, 'ISomeInterface.cs'));
+        interfaceImplUri = vscode.Uri.file(path.join(projectDirectory, 'SomeInterfaceImpl.cs'));
+
+        await vscode.commands.executeCommand("vscode.open", interfaceImplUri);
+        await vscode.commands.executeCommand("vscode.open", interfaceUri);
+    });
+
+    suiteTeardown(async () => {
+        await testAssetWorkspace.cleanupWorkspace();
+    });
+
+    test("When interface is manually renamed, then return correct analysis after re-analysis of project", async function () {
```

Comment:
The code makes some differentiation between "large" and "small" workspaces, do these tests cover those differences?

Reply:
Added tests for large workspace scenario